### PR TITLE
Add new feature: support injecting specified app package name

### PR DESCRIPTION
**Pull request by WindySha**
1. Add a new feature: support injecting by a specified app package name；
2. Optimize the size of output elf file，from 5M to 700K；

### CODE REVIEW

**Comment by erfur** [comment]
Can you please change the target branch to develop?

**Comment by WindySha** [comment]
> Can you please change the target branch to develop?

OK， target branch has been changed to develop.
The reason I added this feature is that my other project [Poros](https://github.com/WindySha/Poros) can use this project as a ptrace alternative.

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "linjector-rs"
-version = "0.1.1-dev"
+version = "0.1.2-dev"
```
Comment (erfur) [inline Cargo.toml:3]:
```suggestion
version = "0.1.1-dev"
```

Version will be bumped on main.

File: README.md [hunk 2]
```diff
@@ -14,7 +14,8 @@ Usage: linjector-cli [OPTIONS] --pid <PID> --file <FILE>
 Options:
   -p, --pid <PID>
           pid of the target process
-
+  -a, --app-package-name <APP_PACKAGE_NAME>
+          target application's package name, restart the application and do injection
```
Comment (erfur) [inline README.md:18]:
```suggestion
          target application's package name, (re)start the application and do injection
```

Could be better to indicate that it works for both cases.

File: src/utils.rs [hunk 3]
```diff
@@ -162,3 +168,82 @@ pub fn fix_file_permissions(file_path: &str) -> Result<(), InjectionError> {
...
+            break;
+        }
+        thread::sleep(Duration::from_micros(500));
+    }
+
+    return _pid;
+}
+
+pub fn restart_app_and_get_pid(pkg_name: &str) -> u32 {
+    let _ = execute_command("am", &vec!["force-stop", "com.ss.android.ugc.aweme"]);
```
[the remark this answers is not in the document]
Comment (WindySha) [reply src/utils.rs:233]: yes，i made a mistake, it should be ‘pkg_name’.

**Pull request merged**